runtime/src/main/java/org/corfudb/runtime/object/VersionLockedObject.java

```diff
@@ -580,7 +582,7 @@ protected void rollbackStreamUnsafe(ISMRStream stream, long rollbackVersion) {
 
         List<SMREntry> entries = stream.current();
 
-        while (entries != null) {
+        while (entries != null && !entries.isEmpty()) {
```

Comment:
You are overriding the empty returned by the existing implementation of `current()` and `previous()`. We need to make sure it doesn't conflict with that.

Reply:
If you want to make this change, then change the lower layers to return empty. Using both null and predicates seem to go against what this PR proposes (i.e. use predicates instead of null)